[music] fix missing lyrics from tag when playing from musicdb url

mkortstiege opened the pull request:
This fixes an issue where the lyrics from tags are not properly shown when playback was started from a `musicdb://` url.

This is caused by using the wrong item path when calling the infotag loader. While at it, I've removed the dupe property handling and injected the lyrics to the current item info tag instead.

http://trac.kodi.tv/ticket/16216

@Montellese, @Paxxi

mkortstiege committed:
[music] fix missing lyrics from tag when playing from musicdb url (fixes #16216)

mkortstiege commented:
jenkins build this please.

Paxxi commented:
This bypasses the database loading entirely so anything downloaded won't show. Can't we load from db as usual and do the extra loading in infomanager::setitem ? That way we won't hit every file on disk and only the ones actually played.

mkortstiege commented:
> This bypasses the database loading entirely so anything downloaded won't show. Can't we load from db as usual and do the extra loading in infomanager::setitem ? That way we won't hit every file on disk and only the ones actually played.

This only happens when the infomanager requests additional info for he currently playing item. The block is only there to fetch the lyrics from the actual tag (not database). Unless i read he code utterly wrong i thikn it's a valid fix.

mkortstiege commented:
Fix is confirmed working. @Montellese comments?

Montellese commented on xbmc/music/MusicInfoLoader.cpp:
```diff
@@ -102,19 +101,21 @@ bool CMusicInfoLoader::LoadAdditionalTagInfo(CFileItem* pItem)
+    // since we're going to load additional tag information we
+    // need to replace the library path with the actual file path.
+    // this has to be done before creating the infotag loader.
+    pItem->SetPath(pItem->GetMusicInfoTag()->GetURL());
```
Do we need to set the path in the `CFileItem` so that the call to `CMusicInfoTagLoaderFactory::CreateLoader` creates the proper loader?

mkortstiege replied:
> Do we need to set the path in the CFileItem so that the call to CMusicInfoTagLoaderFactory::CreateLoader creates the proper loader?

I've changed it to load by item in 3c1145b9a40c88f66c19ab16bf8b74898777b376 which is causing the lyrics issue. What about creating a dummy CFileItem based on the translated path?

Montellese replied:
Maybe blindly using `CFileItem::GetPath()` is not the best approach and we should first check if the `CMusicInfoTag` of the `CFileItem` has a valid path using `CMusicInfoTag::GetURL()` and only fall back to `CFileItem::GetPath()` if it's empty?

mkortstiege commented:
Updated. @Montellese, instead of modifying the actual item pointer i've changed it to create a new item based on the translated path. Also adjusted the comment.

Montellese commented:
I'll just repeat my previous comment:
Maybe blindly using `CFileItem::GetPath()` in `CMusicInfoTagLoaderFactory::CreateLoader` is not the best approach and we should first check if the `CMusicInfoTag` of the `CFileItem` has a valid path using `CMusicInfoTag::GetURL()` and only fall back to `CFileItem::GetPath()` if it's empty?

mkortstiege commented:
It's not blindly using it. `CMusicInfoTagLoaderFactory` is not about the real tag loading only. As for `musicdb://` items it's using the `CMusicInfoTagLoaderDatabase`. In this specific case we need to grab the lyrics from the actual tag since we're not storing them in the library.

Montellese commented:
Ah right I didn't know about `CMusicInfoTagLoaderDatabase` and thought it was simply about getting the proper tag loader based on the file extension.

Montellese commented on xbmc/music/MusicInfoLoader.cpp:
```diff
@@ -107,14 +107,15 @@ bool CMusicInfoLoader::LoadAdditionalTagInfo(CFileItem* pItem)
-  unique_ptr<IMusicInfoTagLoader> pLoader (CMusicInfoTagLoaderFactory::CreateLoader(*pItem));
+  // we load up the actual tag for this file in order to
+  // fetch the lyrics and add it to the current music info tag
+  CFileItemPtr item(new CFileItem(path, false));
```
Maybe name this `tempItem` to make it obvious that i'ts just a temporary? And it also doesn't really need to be a `CFileItemPtr`. A simple `CFileItem` should be enough.

mkortstiege replied:
Sure, will update.

mkortstiege commented:
Updated.

Montellese commented:
Looks good.
